runtime/frame_descriptors.c

```diff
@@ -182,11 +192,16 @@ caml_frame_descrs caml_get_frame_descrs(void)
   CAMLassert(ft);
   if (atomic_load_acq(&ft->free_prev_after_cycle) < caml_major_cycles_completed)
   {
-    /* it's now safe to free the old table */
+    /* it's now safe to free the old table(s) */
     caml_plat_lock(&descr_mutex);
     if (ft->prev != NULL) {
-      caml_stat_free(ft->prev->table.descriptors);
-      caml_stat_free(ft->prev);
+      struct frametable_version *p = ft->prev;
+      while (p != NULL) {
+        struct frametable_version *next = p->prev;
+        caml_stat_free(p->table.descriptors);
+        caml_stat_free(p);
+        p = next;
+      }
       ft->prev = NULL;
       atomic_store_rel(&ft->free_prev_after_cycle, No_need_to_free);
```

Comment:
When reading this part of this code, I was wondering if we want to move this write before the cleaning loop? Since the prev field is not part of the public header and is only modified when holding the `descr_mutex`, it seems fine to me to signal early that there is no cleaning work left to do?